src/udev/udevadm-trigger.c

```diff
@@ -176,6 +176,32 @@ static int device_monitor_handler(sd_device_monitor *m, sd_device *dev, void *us
                 _cleanup_free_ char *saved = NULL;
 
                 saved = set_remove(settle_path_or_ids, syspath);
+                if (!saved) {
+                        const char *old_sysname;
+
+                        /* When the device is renamed, the new name is broadcast, and the old name is saved
+                         * in INTERFACE_OLD. */
+
+                        if (sd_device_get_property_value(dev, "INTERFACE_OLD", &old_sysname) >= 0) {
```

Comment:
hmm, this sounds netdev specific, should we maybe check the subsystem to be "net" first, before interpreting INTERFACE_OLD this way? i think that would be more robust

Reply:
hmm, actually this is not just an issue here: we generate INTERFACE_OLD= based on INTERFACE=, ourselves.

INTERFACE= is not exlcusively emitted by the "net" subsys, but by usb, greybus as well, at least. I think we really should be stricter there and ignore INTERFACE unless subsystem is "net"

Reply:
filed a new bug report about that → https://github.com/systemd/systemd/issues/26156

(after all this has little to do with this PR I guess, but is a general issue of the codebase right now)